Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM jupyter/base-notebook:latest
+
+LABEL maintainer="Fabian Brandt-Tumescheit <brandtfa@hu-berlin.de>"
+
+USER root
+RUN mkdir /var/lib/apt/lists/partial && \
+    apt-get update && \
+    apt-get install -y  --no-install-recommends cmake git build-essential && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+USER 1000
+RUN pip install --upgrade pip setuptools cython
+RUN pip install powerlaw
+RUN pip install sklearn
+RUN pip install seaborn
+RUN pip install pandas
+RUN pip install tabulate
+RUN pip install matplotlib
+RUN pip install networkx
+RUN pip install ipycytoscape
```

Comment:
Docker generates an intermediate image after each COPY or RUN command. Hence, you can avoid a bit of overhead if you merge all of these into a single RUN. Use \ at the end of the line to do a multi-line RUN statement.

Reply:
Apply this to the RUN statements below as well (e.g., by using && to merge several together).

Reply:
This was somewhat intended. It is a trade-off: More commands create smaller overlays, so if anything changes the updates in the image are faster. But since the image is rather static (apart from triggered rebuilds), this makes sense.